python/instrumentation/openinference-instrumentation-langchain/src/openinference/instrumentation/langchain/_tracer.py

```diff
@@ -62,30 +67,73 @@ def audit_timing(wrapped: Any, _: Any, args: Any, kwargs: Any) -> Any:
         print(f"{wrapped.__name__}: {latency_ms:.2f}ms")
 
 
-class _Run(NamedTuple):
-    span: trace_api.Span
-    context: context_api.Context
+K = TypeVar("K", bound=Hashable)
+V = TypeVar("V")
+
+
+class _DictWithLock(ObjectProxy, Generic[K, V]):  # type: ignore
```

Comment:
Why does this cause a type error?

Reply:
This is because `ObjectProxy` from the `wrapt` package is untyped.